katas/main/src/pokerhands/juan/HandtypeChecker.cs

```diff
@@ -0,0 +1,108 @@
+﻿using HandJuan;
+
+namespace HandtypeCheckerJuan
+{
+    public static class HandTypeChecker
+    {
+        public static bool IsStraightFlush(Hand hand)
+        {
+            return IsStraight(hand) && IsFlush(hand);
+        }
+
+        public static bool IsFourOfAKind(Hand hand)
+        {
+            if (hand.GroupsAttr.Count() == 2)
+            {
+                return (
+                    hand.GroupsAttr.ElementAt(0).Count() == 4
+                    || hand.GroupsAttr.ElementAt(0).Count() == 1
+                );
+            }
+
+            return false;
+        }
+
+        public static bool IsFullHouse(Hand hand)
+        {
+            if (hand.GroupsAttr.Count() == 2)
+            {
+                return (
+                    hand.GroupsAttr.ElementAt(0).Count() == 2
+                    || hand.GroupsAttr.ElementAt(0).Count() == 3
+                );
+            }
+
+            return false;
+        }
+
+        public static bool IsFlush(Hand hand)
+        {
+            for (int i = 1; i <= 4; i++)
```

Comment:
try to avoid single letter names for variables, even iterators, it is more readable to have `index` instead of `i`